pom.xml

```diff
@@ -165,6 +165,11 @@
                 <version>0.20.6</version>
             </dependency>
             <dependency>
+                <groupId>org.jetbrains</groupId>
+                <artifactId>annotations</artifactId>
+                <version>15.0</version>
```

Comment:
Can yoou set "provided" here and have it propagated?

Reply:
i used to do that but it violates the principle of least surprise:
someone adding the dependency as
```
<dependency>
  <groupId>...</groupId>
  <artifactId>...</artifactId>
</dependency>
```
and expecting it to get (the default) `<scope>compile</scope>` would get the overridden scope (`provided`, `test`, etc.) instead.

Reply:
I see. Whoever does use the version should check their effective pom in any case.
If this pattern is used throughout - then it is fine.